prime-router/docs/proposals/0019-website-api-payload-structure.md

```diff
@@ -0,0 +1,332 @@
+                        "matches(ordering_facility_state, IG)"
+                    ],
+                    "qualityFilter": null,
+                    "routingFilter": null,
+                    "processingModeFilter": null,
+                    "conditionFilter": null
+                },
+                ...
+            ],
+            "featureFlags": [],
+            "keys": []
+        },
+        ...
+    ]
+}
+```
+
+#### Paginated collections
+
+Paginated collection request paths should take the form of `*/<resources>?page=<page>&pageSize=<pageSize>`, where:
+
+- `page` is the requested page (e.g., `1`, `2`, `10`)
+- `pageSize` is the size of the request pages (e.g., `10`, `20`, `50`)
+
+Paginated collection responses should take the form of:
+
+```json
+{
+    "meta": {
+        "type": <resource_type>,
+        "totalCount": <total_count>,
+        
+        // pagination-specific values
+        "totalPages": <count_of_total_pages>,
+        "previousPage": <previous_page_number>,
+        "nextPage": <next_page_number>,
+    },
+    "data": [
+        {
+            "property1": ...,
+            "property2": ...,
+            "property3": ...,
+            // other resource data
+            ...
+        },
+        // other resource objects
+        ...
+    ]
+}
+```
+
+The `meta` for paginated collection responses should include the following values:
+
+- `totalPages` is the count of total pages
+- `previousPage` is the previous page number (`null` if on the first page)
+- `nextPage` is the next page number (`null` if on the last page)
+
+##### Example
+
+`GET /api/settings/organizations?page=1&pageSize=50`
+
+```json
+{
+    "meta": {
+        "type": "Organization",
+        "totalCount": 100,
+
+        // pagination-specific values
+        "totalPages": 2,      // (100 / 50 = 2)
+        "previousPage": null, // on first page so previous page is null 
+        "nextPage": 2
+    },
+
+    "data": [
+        {
+            "name": "ignore",
+            "description": "FOR TESTING ONLY",
+            "jurisdiction": "FEDERAL",
+            "filters": [
+                {
+                    "topic": "covid-19",
+                    "jurisdictionalFilter": [
+                        "matches(ordering_facility_state, IG)"
+                    ],
+                    "qualityFilter": null,
+                    "routingFilter": null,
+                    "processingModeFilter": null,
+                    "conditionFilter": null
+                },
+                ...
+            ],
+            "featureFlags": [],
+            "keys": []
+        },
+        // other resource objects
+        ...
+    ]
+}
+```
+
+#### Sorted collections
+
+Sorted collection request paths should take the form of `*/<resources>?sortDir=<sortDir>&sortColumn=<sortColumn>`, where:
+
+- `sortDir` is the order of sorting (`ASC` or `DESC`)
+- `sortColumn` is the name of the sorted column (e.g., `id`, `name`, `expiresAt`)
```

Comment:
Is there potential for a need to accept multiple columns for sortColumn in the future? Or any property for that matter. Should we establish the the syntax rules for this scenario (ex: comma separated)?

Reply:
Great point, I think we should use comma-separated values for this with descending priority -- will update the proposal!  Thanks for calling this out!